lang/cs/Org.Apache.REEF.Network.Examples/GroupCommunication/BroadcastReduceDriverAndTasks/BroadcastReduceDriver.cs

```diff
@@ -111,12 +116,14 @@ public void OnNext(IEvaluatorRequestor evaluatorRequestor)
         public void OnNext(IAllocatedEvaluator allocatedEvaluator)
         {
             IConfiguration contextConf = _groupCommDriver.GetContextConfiguration();
-            IConfiguration serviceConf = _groupCommDriver.GetServiceConfiguration();
+            IConfiguration serviceConf = _groupCommDriver.GetServiceConfiguration(_codecConfig);
```

Comment:
You should call GetServiceConfiguration() without parameter and then merge it with _condecConfig at this line. In this way, GetServiceConfiguration() in the GroupCommDriver would only supply system level configurations. It is up to teh clients what you would like to add on without changing GroupCommDriver API.